roles/hcl/docs/setup-environment/tasks/main.yml

```diff
@@ -6,7 +6,7 @@
   include_tasks:           setup_os.yml
   when:
     - __setup_environment|bool
-    - inventory_hostname in groups["docs_servers"] or inventory_hostname in groups["conversion_servers"] or inventory_hostname in groups["viewer_servers"] or inventory_hostname in groups["proxy_servers"]
+    - inventory_hostname in groups["docs_servers"] or inventory_hostname in groups["conversion_servers"] or inventory_hostname in groups["viewer_servers"] or inventory_hostname in groups["proxy_servers"]  or inventory_hostname in groups["dmgr"]
```

Comment:
I see the error is when installing the Docs Editor Extension, in this case should it be `groups["cnx_was_servers"][0]` since the installer is run there?
https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/84b57ec298438efde896682b41c1d89d48b90ba8/roles/hcl/docs/tasks/main.yml#L59

Reply:
In my environment the dmgr and connections server are on the same host (VM), but the dmgr server could be on a separate host than the (first) connections server.

The Docs documentation says about the pre-installation tasks for [Installation Python](https://help.hcltechsw.com/docs/v2.0.2/onprem/install_guide/guide/text/installing_python.html)
> **Installing Python**
>
>The HCL Docs deployment requires a Python 3 runtime environment. For both Windows and Linux installations, ensure >that Python is installed on the Connections _**Deployment Manager**_ and on all Linux™ servers hosting an File Viewer, HCL >Docs, or HCL Docs proxy node.

With my customization the playbook finished successfully.

Reply:
We have test env with Connections and Docs applications in one box and some where Connections and Docs applications are on separate boxes.

Reply:
`
With your decision "we made a decision to use the first Connections server groups["cnx_was_servers"][0]. " in mind this part or inventory_hostname in groups["cnx_was_servers"][0] should be added in the last line of the when clause .
`
Yes, exactly.  That's all I'm suggesting in https://github.com/HCL-TECH-SOFTWARE/connections-automation/pull/309#discussion_r1282468727.  

We know that the Extension installer is set to run on `groups["cnx_was_servers"][0]` per
https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/84b57ec298438efde896682b41c1d89d48b90ba8/roles/hcl/docs/tasks/main.yml#L59 

Scenarios:
a) When Dmgr and groups["cnx_was_servers"][0] are different VMs   (eg. hosts `dmgr.example.com`, `connections1.example.com`)
--> in this case, group `[dmgr]` in inventory.ini will have `dmgr.example.com` and group`[cnx_was_servers]` will have `connections1.example.com`.  Since [L59 above](https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/84b57ec298438efde896682b41c1d89d48b90ba8/roles/hcl/docs/tasks/main.yml#L59 ) says to run on `groups["cnx_was_servers"][0]`, the installer will run on `connections1.example.com` and it will work if setup_os is run there. 

b) When Dmgr and groups["cnx_was_servers"][0] are the same VM  (eg. host `Connections1`)
--> in this case, group `[dmgr]` and group`[cnx_was_servers]` in inventory.ini will both set to `connections1.example.com`, as such the installer will still run on `connections1.example.com` and it will work if setup_os is run there. 

---------
Re: `But I don't know if this enhancement will work in a production environment where the dmgr server can be on another server than the (first) Connections server. This must be tested.`
If  [L59 above](https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/84b57ec298438efde896682b41c1d89d48b90ba8/roles/hcl/docs/tasks/main.yml#L59 ) is fixed to run the installer on `groups["cnx_was_servers"][0]`.  Is the possibility of dmgr not on the (first) Connections server a concern?